src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/Routing/ActionEndpointFactory.cs

```diff
@@ -9,32 +9,25 @@
 using Microsoft.AspNetCore.Mvc.Abstractions;
 using Microsoft.AspNetCore.Mvc.ActionConstraints;
 using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
 using Microsoft.AspNetCore.Routing;
 using Microsoft.AspNetCore.Routing.Patterns;
+using Microsoft.Extensions.DependencyInjection;
 
 namespace Microsoft.AspNetCore.Mvc.Routing
 {
     internal class ActionEndpointFactory
     {
         private readonly RoutePatternTransformer _routePatternTransformer;
-        private readonly MvcEndpointInvokerFactory _invokerFactory;
 
-        public ActionEndpointFactory(
-            RoutePatternTransformer routePatternTransformer,
-            MvcEndpointInvokerFactory invokerFactory)
+        public ActionEndpointFactory(RoutePatternTransformer routePatternTransformer)
```

Comment:
This is a lot better now because we don't need to create a bunch of state/services wicked early. There's an extra DI lookup on the hot path, but if that's a problem we can mitigate that in a few ways.

Reply:
What is the perf impact of `var invokerFactory = context.RequestServices.GetRequiredService<IActionInvokerFactory>();`? Can we measure and fix it as part of this PR or create a followup issue?

Reply:
I'll make the change to amortize this.